File: backend/src/main/java/de/hemfeinkost/backend/services/abstracts/CategoryService.java
package de.hemfeinkost.backend.services.abstracts;

import de.hemfeinkost.backend.services.dtos.requests.CategoryRequest;
import de.hemfeinkost.backend.services.dtos.responses.CategoryCreatedResponse;
import de.hemfeinkost.backend.services.dtos.responses.CategoryGetAllResponse;

import java.util.List;

public interface CategoryService {
    List<CategoryGetAllResponse> getAllCategories();
    List<CategoryGetAllResponse> getActiveCategories();
    CategoryCreatedResponse getCategoryById(long id);
    CategoryCreatedResponse addCategory(CategoryRequest categoryRequest);
    CategoryCreatedResponse updateCategory(long id, CategoryRequest categoryRequest);
    CategoryCreatedResponse changeCategoryStatus(long id, boolean status);
}

File: frontend/src/services/CategoryService.ts
import axios from "axios";
import {Category} from "../types/Category.ts";

export default class CategoryService {
    getAllCategories() {
        return axios.get('/api/categories')
    }

    getActiveCategories() {
        return axios.get('/api/categories/active')
    }

    getCategoryById(id: number) {
        return axios.get(`/api/categories/${id}`)
    }

    addCategory(category: Category) {
        return axios.post('/api/categories', category)
    }

    updateCategory(id: number, category: Category) {
        return axios.put(`/api/categories/${id}`, category)
    }

    changeCategoryStatus(id: number, status: boolean) {
        return axios.put(`/api/categories/status/${id}?status=${status}`)
    }
}